lib/tree.rb

```diff
@@ -0,0 +1,87 @@
+
+# require 'pry-byebug'
+
+Node = Struct.new(:attributes, :children)
```

Comment:
A bit redundant to have children here and in the tag. Probably children and parent should be properties of the Node and everything else a property of the Tag.